### PR TITLE
Feat/avatar

### PR DESCRIPTION
**Pull request by jamesducky**
Closes [DEVELOP: Avatar](https://app.asana.com/0/1198920871936085/1202312799011611/f)

### Summary of changes included in this PR
- Create `avatar` component
- Updated example React app with two `avatar` examples: interactive(clickable) and non-interactive
- Story book examples
- Tests

### Reviewer checklist:

- Tests for
  - accessibility
  - rendered HTML (spec)
- Storybook stories for all variants
- JSDoc documentation for component
- Update Example app (React, ...)

### CODE REVIEW

File: examples/react-app/src/Avatars.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        ></PlmgAvatar>
+        <PlmgAvatar style={styles} size={'medium'}></PlmgAvatar>
+        <PlmgAvatar style={styles} size={'medium'} userDeleted></PlmgAvatar>
+        <PlmgAvatar
+          size={'medium'}
+          style={styles}
+          imageUrl="https://static.ducky.eco/icons/maskable_icon_192.png"
+        ></PlmgAvatar>
+        <PlmgAvatar style={styles} size={'medium'}></PlmgAvatar>
+        <PlmgAvatar size={'medium'} style={styles} userDeleted></PlmgAvatar>
```
Comment (nathan-bird) [inline examples/react-app/src/Avatars.tsx:24]: Imagine the use case where these avatars would be grouped. It is unlikely we will define each one rather will have a list.
I suggest making a profiles object which contains information on users. Then have an array of avatars you'd like to show. Map that array drawing information from your profiles object.

Comment (nathan-bird) [inline examples/react-app/src/Avatars.tsx:24]: How could we programatically define if a user is deleted rather than explicitly saying they are? Is there something we can check the existence of?

File: examples/react-app/src/index.tsx [hunk 2]
```diff
@@ -76,18 +78,28 @@ ReactDOM.render(
...
           flexDirection: 'row',
           justifyContent: 'space-between',
+          alignItems: 'center',
         }}
       >
         <span>Menu Item</span>
+
         <PlmgSeparator
           thickness="thick"
+          style={{ alignSelf: 'stretch' }}
```
Comment (nathan-bird) [inline examples/react-app/src/index.tsx:88]: Why did you add this style? Is it required for the component to work? If it is you should make a bug report so it is fixed for all implementations rather than only in this app.

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.stories.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    'https://static.ducky.eco/icons/maskable_icon_192.png',
+    undefined,
+  ];
+
+  let htmlContent = '';
+  imageUrl.forEach((image) => {
+    userDeleted.forEach((deleted) => {
+      sizes.forEach((size) => {
+        if (image) {
+          htmlContent += `<plmg-avatar interactive image-url=${image} size=${size}></plmg-avatar>`;
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-avatar/plmg-avatar.stories.js:82]: Your 'All Variations' story are all intractable. I would expect it show show non-interactive avatars too.

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.stories.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        } else {
+          htmlContent += `<plmg-avatar interactive user-deleted=${deleted} size=${size}></plmg-avatar>`;
+        }
+      });
+    });
+  });
+
+  console.log(htmlContent);
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-avatar/plmg-avatar.stories.js:90]: Remove this console log.

**Comment by JohannesAnd** [comment]
I don't know much about web components, so I don't want to accept or reject this, but I had a few thoughts:
1. The prop `label` seems to be missing from Storybook
2. The userDeleted prop could also be an icon prop, which would allow any icon to be used

**Comment by jamesducky** [comment]
Thanks very much for taking a look

> 1. The prop `label` seems to be missing from Storybook
Cheers for picking that up. Added in now. 

> 2. The userDeleted prop could also be an icon prop, which would allow any icon to be used

Good suggestion. Would make the component more useful, but might be outside of the current design. So instead of a boolean have a 'userDeletedIcon' prop that would accept an icon name string?

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+// Component style
+// -------------------------------------------------
+
+.plmg-avatar {
+  background-color: tokens.$plmg-color-background-neutral-medium;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  color: tokens.$plmg-color-icon-neutral;
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-avatar/plmg-avatar.scss:23]: The spec says:
> The background color, icon color is not fixed and can be modified

Please update.

**Review by nathan-bird — CHANGES_REQUESTED** [review]

**Comment by nathan-bird** [comment]
> > 3\. he userDeleted prop could also be an icon prop, which would allow any icon to be used
> 
> Good suggestion. Would make the component more useful, but might be outside of the current design. So instead of a boolean have a 'userDeletedIcon' prop that would accept an icon name string?

I agree this is outside the design spec and may cause dev confusion or design mismatch if the wrong icon prop is passed.

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.scss (hunk 5 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-avatar/plmg-avatar.scss:23]: So two extra props: `iconColor` and `backgroundColor` (if not passed, colors set to the defaults)?

File: packages/component-library/src/components/plmg-avatar/test/plmg-avatar.spec.tsx [hunk 6]
```diff
@@ -49,6 +49,37 @@ describe('plmg-avatar', () => {
       </plmg-avatar>
     `);
   });
+  it('renders without user defined colors when an image is provided', async () => {
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-avatar/test/plmg-avatar.spec.tsx:52]: This is just a thought; I haven't tested it. Can we test the presence and value of a css property? 
I wonder if we can get access to our styles using [`window.getComputedStyle(el)`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle)
Could you have a look into it?

Comment (jamesducky) [reply packages/component-library/src/components/plmg-avatar/test/plmg-avatar.spec.tsx:52]: Yep. That works. I've committed two tests that check the background color is applied conditionally. Tried to write a test to check the color prop on the <plmg-svg-icon> component is working, but I can't select the svg element inside .... yet - further research needed!

File: examples/react-app/src/Avatars.tsx (hunk 1 above)
Comment (jamesducky) [reply examples/react-app/src/Avatars.tsx]: I've implemented your suggestion using a simple object (I considered calling a test API of users to get a more realistic complicated json object - but thought that might be too much clutter for the example)

Comment (jamesducky) [reply examples/react-app/src/Avatars.tsx]: I've used an empty user name as the condition. Again, I imagine that it is more likely that the object would contain a user_deleted key or timestamp - but I used the lack of user name as a proxy - as I suppose all personal information would be deleted.

File: examples/react-app/src/Avatars.tsx [hunk 7]
```diff
@@ -1,28 +1,55 @@
...
+  users.map(({ name, picture, user_id }) => (
+    <PlmgAvatar
+      key={user_id}
+      userDeleted={name === '' ? true : false}
+      imageUrl={picture}
+      size={'medium'}
+      style={user_id !== '1' ? { marginLeft: '-15px' } : null}
+    />
+  )),
+];
```
Comment (JohannesAnd) [inline examples/react-app/src/Avatars.tsx:46]: I know this isn't production code per se, so it is not critical:
1. This results in a nested array, so `[[<Component/>, <Component/>]]`, I'd get rid of the surrounding array
2. `name === '' ? true : false` is the same as `name === ''`
3. Rather than marginLeft, I think we would get the same result with `gap: -15px` on the flex div and avoid the ternary

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    const backgroundColorStyle = {
+      backgroundColor: this.backgroundColor ?? plmgColorBackgroundNeutralMedium,
+    };
+
+    return (
+      <div
+        tabIndex={this.interactive ? 0 : -1}
+        onClick={this.interactive ? (e) => this.avatarClick.emit(e) : null}
+        class={classes}
+        style={!this.imageUrl && backgroundColorStyle}
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-avatar/plmg-avatar.tsx:142]: If there is an `imagerUrl` and `userDeleted` no default background colour is set.

**Review by nathan-bird — CHANGES_REQUESTED** [review]

File: examples/react-app/src/Avatars.tsx (hunk 7 above)
Comment (jamesducky) [reply examples/react-app/src/Avatars.tsx]: 1, 2 - great. thanks for corrections
3 - gap won't take a negative value. I've taken out the ternary and added a margin-left on the container div to negate the positioning of the negative margin on the first flex item.

File: packages/component-library/src/components/plmg-avatar/plmg-avatar.tsx (hunk 8 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-avatar/plmg-avatar.tsx]: Fixed and added a test to check background color is applied.

File: examples/react-app/src/Avatars.tsx (hunk 7 above)
Comment (JohannesAnd) [reply examples/react-app/src/Avatars.tsx]: You're absolutely right, my bad

**Review by nathan-bird — APPROVED** [review]
